File: src/main/java/com/sorsix/eventagregator/service/InviteService.java
package com.sorsix.eventagregator.service;

import com.sorsix.eventagregator.model.DTO.InvitationDTO;
import com.sorsix.eventagregator.model.Invitation;

import java.util.Optional;

public interface InviteService {

    Optional<Invitation> invite(InvitationDTO invitationDTO);

}

File: event-aggregator-frontend/src/app/services/invite.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs/Rx";
import {InviteDTO} from "../model/InvitationDTO";

@Injectable({
  providedIn: 'root'
})
export class InviteService {

  constructor(private http: HttpClient) { }

  public sendInvitation(invitation: InviteDTO): Observable<any> {
    return this.http.post("/api/invite", invitation);
  }

}
